app/controllers/films_controller.rb

```diff
@@ -1,16 +1,43 @@
 require 'pry'
+require 'open-uri'
 class FilmsController < ApplicationController
+  def add_movie
+    movie_id = params[:movie_id]
+      movie = Film.create(
+        rapid_id: params['movie_id'],
+        title: params['title'],
+        rating: params['rating'],
+        genre: params['genre'],
+        year: params['year']
+      )
+    if movie.save
+      if params['picture'].present?
```

Comment:
```
# mVoie Save without pitcture 
if params['picture'].present? #NO else need here it will only check if picture is comming need to save or not
  if images save
     # If Child Record Save 
  else
     # Imge save error 
  end
end
```

Reply:
def add_movie
    movie = Film.create(
      rapid_id: params['movie_id'],
      title: params['title'],
      rating: params['rating'],
      genre: params['genre'],
      year: params['year']
    )
    if movie.save
      if params['picture'].present?
        movie.images.create(remote_picture_url: params['picture'])
        redirect_to 'index.html.erb', notice: 'Movie added to the database.'
      else
        redirect_to 'index.html.erb', notice: 'Movie added to the database without a picture.'
      end
    else
      redirect_to 'index.html.erb',"Error: #{movie.errors.full_messages.join(", ")}"
    end
  end